packages/lit-node-client-nodejs/src/lib/lit-node-client-nodejs.ts

```diff
@@ -495,18 +497,66 @@ export class LitNodeClientNodeJs
    * will call refresh if the block hash is expired
    * @returns {Promise<string>} latest block hash from `handhsake` with the lit network.
    */
-  getLatestBlockhash = (): string => {
-    if (!this.ready) {
-      logError('Client not connected, remember to call connect');
+  getLatestBlockhash = async (): Promise<string> => {
+    const requestId = this.getRequestId();
+
+    this.connectedNodes = new Set();
+
+    if (this.config.bootstrapUrls.length <= 0) {
       throwError({
-        message: 'Client not connected',
-        errorKind: LIT_ERROR.LIT_NODE_CLIENT_NOT_READY_ERROR.kind,
-        errorCode: LIT_ERROR.LIT_NODE_CLIENT_NOT_READY_ERROR.code,
+        message: `Failed to get bootstrapUrls for network ${this.config.litNetwork}`,
+        errorKind: LIT_ERROR.INIT_ERROR.kind,
+        errorCode: LIT_ERROR.INIT_ERROR.name,
+      });
+    }
+
+    const respList = [];
+
+    for (const url of this.config.bootstrapUrls) {
+      const challenge = this.getRandomHexString(64);
+
+      try {
+        const resp = await this.handshakeWithNode(
+          { url, challenge },
+          requestId
+        );
+        respList.push(resp);
+      } catch (e) {
+        logError(`Error getting latestBlockhash from node ${url}`, e);
+      }
+    }
+
+    // -- if the list is empty, then return an error
+    if (respList.length === 0) {
+      throwError({
+        message: 'Failed to get latestBlockhash from any node',
+        errorKind: LIT_ERROR.INIT_ERROR.kind,
+        errorCode: LIT_ERROR.INIT_ERROR.name,
       });
     }
 
-    // we are confident in this value being non null so we return
-    return this.latestBlockhash!;
+    // -- pick the latestBlockhash the appears the most
+    const blockhashCounts = respList.reduce(
+      (acc: { [key: string]: number }, response) => {
+        const { latestBlockhash } = response;
+
+        // Check if latestBlockhash is defined
+        if (latestBlockhash) {
+          acc[latestBlockhash] = (acc[latestBlockhash] || 0) + 1;
+        }
+
+        return acc;
+      },
+      {}
+    );
+
+    // -- get the most frequent blockhash
+    const mostFrequestBlockhash = Object.keys(blockhashCounts).reduce(
+      (a: string, b: string) =>
+        blockhashCounts[a] > blockhashCounts[b] ? a : b
+    );
+
+    return mostFrequestBlockhash;
```

Comment:
Let's use Lodash here?

Reply:
i think we have this functionality in here: https://github.com/LIT-Protocol/js-sdk/blob/8c0a098c75cf3c6949718e5d0fb2cfaa0970182d/packages/misc/src/lib/misc.ts#L60

should we use this instead?

Reply:
@Adarsh-Kumar28 I wasn't sure if we should introduce a new library particularly for this function. I would like to have a new PR/task dedicated to replace everything that can be refactored to use Lodash instead. What's your opinion on this @MaximusHaximus

Reply:
> @Adarsh-Kumar28 I wasn't sure if we should introduce a new library particularly for this function. I would like to have a new PR/task dedicated to replace everything that can be refactored to use Lodash instead. What's your opinion on this @MaximusHaximus

@Adarsh-Kumar28  I wasn't sure if we should introduce a new library particularly for this function. I would like to have a new PR/task dedicated to replace everything that can be refactored to use Lodash instead. What's your opinion on this @MaximusHaximus

Reply:
> Let's use Lodash here?

@Adarsh-Kumar28 I wasn't sure if we should introduce a new library particularly for this function. I would like to have a new PR/task dedicated to replace everything that can be refactored to use Lodash instead. What's your opinion on this @MaximusHaximus



> i think we have this functionality in here:
> 
> https://github.com/LIT-Protocol/js-sdk/blob/8c0a098c75cf3c6949718e5d0fb2cfaa0970182d/packages/misc/src/lib/misc.ts#L60
> 
> should we use this instead?

Yea, we can for the `mostFrequestBlockhash` constant, but we still need to use the .reduce of extract all blockhash as array.